bpaotu/bpaotu/views.py

```diff
@@ -555,6 +560,44 @@ def yield_csv_function():
     return response
 
 
+@csrf_exempt
+def otu_verification_endpoint(request):
+    try:
+        data = request.GET.get('result')
+
+        hash_portion = data.split('||')[0]
+        data_portion = data.split('||')[1]
+
+        json_data = json.loads(data_portion)
+
+        timestamp = json_data['timestamp']
+        organisations = json_data['organisations']
+    except:
+        return HttpResponse('There is an error with the authentication token.')
+
+    secret_key = bytes(os.environ.get('BPAOTU_AUTH_SECRET_KEY'), encoding='utf-8')
+
+    digest_maker = hmac.new(secret_key)
+    digest_maker.update(data_portion.encode('utf8'))
+    digest = digest_maker.hexdigest()
+
+    SECS_IN_DAY = 60*60*24
+    response = ''
+
+    if digest == hash_portion:
+        if time.time() - timestamp < SECS_IN_DAY:
+            if 'australian-microbiome' in organisations:
+                response = "Valid"
+            else:
+                response = "You do not have access to the Ausmicro data."
+        else:
+            response = "The timestamp is too old."
+    else:
+        response = "Secret key does not match."
+
+    return HttpResponse(response)
+
```

Comment:
this approach of returning strings is wrong - use HTTP status codes, they're well defined and machine understandable.

break this out into a security check function that raises the Http403 status, and then call it from all of the API endpoints.

Reply:
Fixed.